netket/experimental/operator/_fermions_2nd.py

```diff
@@ -669,10 +669,34 @@ def _check_hermitian(
     # check if hermitian by comparing the dictionaries
     dict_normal = dict(dict_normal)
     dict_hc_normal = dict(dict_hc_normal)
-    is_hermitian = dict_normal == dict_hc_normal
+
+    # compare dict up to a tolerance
+    is_hermitian = dict_compare(dict_normal, dict_hc_normal)
+
     return is_hermitian
 
 
+def dict_compare(d1, d2, rtol=1e-05, atol=1e-08, equal_nan=False):
+    """Compare two dicts and return True if their keys and values are all the same (up to some tolerance)"""
+    d1_keys = set(d1.keys())
+    d2_keys = set(d2.keys())
+    shared_keys = d1_keys.intersection(d2_keys)
+    added = d1_keys - d2_keys
+    if len(added) != 0:
+        return False
+    removed = d2_keys - d1_keys
+    if len(removed) != 0:
+        return False
```

Comment:
Can't you just do the more readable
```
if not d1_keys == d2_keys:
    return False
```
?
Or this does not work for some reason?